libs/librrgraph/src/base/rr_graph_view.h

```diff
@@ -645,4 +757,11 @@ class RRGraphView {
     const vtr::vector<RRSegmentId, t_segment_inf>& rr_segments_;
     /// switch info for rr nodes
     const vtr::vector<RRSwitchId, t_rr_switch_inf>& rr_switch_inf_;
+
+    /** A list of incoming edges for each routing resource node. This can be built optionally, as required by applications.
+     *  By default, it is empty! Call build_in_edges() to construct it!!! */
+    const vtr::vector<RRNodeId, std::vector<RREdgeId>>& node_in_edges_;
+
+    /** A list of extra ptc numbers for each routing resource node. See details in RRGraphBuilder class */
```

Comment:
Rename to node_tileable_track_nums.
Say this is only loaded for tilable rr-graphs.
Change implementation to check if it is loaded before set/get operations.

Longer term: maybe we can store this more efficiently (it is a modolo computation, so I don't think we need to store it on every location along a wire).